core/source/api/opencmiss/zinc/scene.h

```diff
@@ -398,6 +415,102 @@ ZINC_API int cmzn_scene_get_spectrum_data_range(cmzn_scene_id scene,
+ *
+ * @param scene  The scene to modify.
+ * @param transformation_field  A 16 component field whose values give a 4x4
+ * transformation of homogeneous coordinates, or NULL/invalid to clear. This
+ * field must be spatially constant, but it may be time-varying in which case
+ * scene redraws are automatically trigger when the time changes. Runtime
+ * errors will be reported if this field cannot be evaluated given only the
+ * current timekeeper time. Field must belong to this scene's owning region.
+ * Alternatively, iff a transformation field is in use, passing an invalid/NULL
+ * field to this function clears transformation.
+ * @return  Result OK on success, any other value on failure.
+ */
+ZINC_API int cmzn_scene_set_transformation_field(cmzn_scene_id scene,
+	cmzn_field_id transformation_field);
+
+/**
+ * Get 4x4 transformation matrix of the scene local coordinates into the
+ * parent scene local coordinate system, or world coordinate system if scene is
+ * for the root region. If a transformation field is active, calculates the
+ * current transformation matrix from the field, or returns an error if the
+ * field cannot be evaluated. If no transformation is active this function
+ * returns the 4x4 identity matrix.
+ * @see cmzn_scene_set_transformation_matrix
+ * @see cmzn_scene_set_transformation_field
+ *
+ * @param scene  The scene to query.
+ * @param valuesCount  The size of the valuesOut array, at least 16.
+ * @param valuesOut  An array of at least size given by valuesCount to fill
+ * with the values of the transformation matrix. If valuesCount exceeds the
+ * number of components in the transformation matrix, these are not set.
+ * @return  Result OK on success, any other value on failure.
+ */
+ZINC_API int cmzn_scene_get_transformation_matrix(cmzn_scene_id scene,
+	int valuesCount, double *valuesOut);
+
+/**
+ * Set fixed 4x4 transformation matrix of the scene local coordinates into the
+ * parent scene local coordinate system, or world coordinate system if scene is
+ * for the root region. Clears any current scene transformation field.
+ *
+ * The 16 components give the following matrix transformation:
+ * <pre>
+ *     |x.out| = | c01 c02 c03 c04 | |x.in|
+ *     |y.out| = | c05 c05 c07 c08 | |y.in|
+ *     |z.out| = | c09 c10 c11 c12 | |z.in|
+ *     |h.out| = | c13 c14 c15 c16 | |h.in|
+ * </pre>
+ * where 'in' coordinates are local coordinates in the scene, and 'out'
+ * coordinates are in the parent local or world coordinate system.
+ * Homogeneous coordinates are used after division by the h component which
+ * gives a perspective effect if not equal to 1.
+ * Typical no-perspective usage uses h.in=1 and the bottom row 0 0 0 1 so
+ * h.out=1, hence c04, c08, c12 give displacement and the upper-left 3x3 matrix
+ * defines basis vectors of local coordinates in terms of parent coordinates.
+ *
+ * @param scene  The scene to modify.
+ * @param valuesCount  The size of the valuesIn array, currently 16.
```

Comment:
I initially wrote this thinking we may also support 9-component (rotation-scale-only) and 12 component (no perspective) variants, but that's just a convenient way of padding values which you can do with fields already (without complicating the scene code), so I think it's not needed. These transformation matrices are really just a legacy, if useful, feature... if you want to do something special these days you'll use a vertex shader.
So I'll ditch the valuesCount and force it to be 16. OK?